lib/config.js

```diff
@@ -133,6 +133,8 @@ var PromisesDependency;
  *       failing to establish a connection with the server after
  *       `connectTimeout` milliseconds. This timeout has no effect once a socket
  *       connection has been established.
+ *     * **lookup** [Function] &mdash; Custom DNS lookup function. This feature is
+ *       only supported in the Node.js (>= v12.x) environment.
```

Comment:
The option lookup was added in Node,js v2.x in https://github.com/nodejs/node/commit/4abe2fa1cfcc434952570c1c979dd4ce150fba67

Only the documentation was updated in https://github.com/nodejs/node/commit/5bd6f516d82d069ff8710b86108dedc333b2b580

```suggestion
 *     * **lookup** [Function] &mdash; Custom DNS lookup function.
```

Instead of making additional commits, the following commits can be reverted:
* [`0894018` (#3524)](https://github.com/aws/aws-sdk-js/pull/3524/commits/089401836f35c83e0fb157f18776a78bd758c127)
* [`adb765c` (#3524)](https://github.com/aws/aws-sdk-js/pull/3524/commits/adb765c0ac1f149c8ef8dc6902e914ff1c415e58)
* [`276eb80` (#3524)](https://github.com/aws/aws-sdk-js/pull/3524/commits/276eb80c903d5bd41a60603d98adcc925cc72b9f)